src/bp-activity/bp-activity-functions.php

```diff
@@ -311,7 +311,114 @@ function bp_activity_get_userid_from_mentionname( $mentionname ) {
+ *                                     Default ['activity'].
+ *     @type integer  $position        The order of the activity type in front-end dropdowns.
+ *                                     Default 0.
+ *     @type array    $supports        The list of supported features for the activity type.
+ *                                     Default [].
+ * }
+ * @return BP_Activity_Type|WP_Error The registered activity type object on success,
+ *                                   WP_Error object on failure.
+ */
+function bp_register_activity_type( $type, $args = array() ) {
+	$activity_types = buddypress()->activity->types;
+
+	// Sanitize post type name.
+	$type = sanitize_key( $type );
+
+	if ( empty( $type ) || strlen( $type ) > 20 ) {
+		_doing_it_wrong( __FUNCTION__, __( 'BP Activity Type names must be between 1 and 20 characters in length.', 'buddypress' ), '14.0.0' );
+		return new WP_Error( 'activity_type_length_invalid', __( 'BP Activity Type names must be between 1 and 20 characters in length.', 'buddypress' ) );
+	}
+
+	if ( isset( $activity_types[ $type ] ) ) {
+		return new WP_Error( 'activity_type_already_registered', __( 'BP Activity Types are unique. The type you want to register already exists.', 'buddypress' ) );
+	}
+
+	$type_object                           = new BP_Activity_Type( $type, $args );
+	buddypress()->activity->types[ $type ] = $type_object;
+
+	return $type_object;
+}
+
+/**
+ * Unregisters an activity type.
+ *
+ * @since 14.0.0
+ *
+ * @param  string $type The name key of the activity type.
+ * @return bool|WP_Error The registered activity type object on success,
+ *                                   WP_Error object on failure.
+ */
+function bp_unregister_activity_type( $type ) {
+	$type_object = bp_get_activity_type_object( $type );
+
+	if ( is_wp_error( $type_object ) ) {
+		return $type_object;
+	}
+
+	unset( buddypress()->activity->types[ $type_object->name ] );
+
+	return true;
+}
+
+/**
+ * Retrieves the Activity type object for a specific type.
+ *
+ * @since 14.0.0
+ *
+ * @param string $type The name key of the activity type.
+ * @return WP_Error|BP_Activity_Type The Activity type object.
+ */
+function bp_get_activity_type_object( $type ) {
+	$activity_types = buddypress()->activity->types;
+
+	// Sanitize activity type name.
+	$type = sanitize_key( $type );
+
+	if ( ! isset( $activity_types[ $type ] ) ) {
+		return new WP_Error( 'invalid_activity_type', __( 'Invalid activity type.', 'buddypress' ) );
+	}
+
+	return $activity_types[ $type ];
+}
+
+/**
+ * Retrieves the list of Activity types having a specific role.
+ *
+ * @since 14.0.0
+ *
+ * @param string $role The requested role for activity types to retrieve.
+ * @return array The list of Activity type name keys having the requested role.
+ */
+function bp_get_activity_types_for_role( $role ) {
+	$types = wp_filter_object_list( buddypress()->activity->types, array( 'role' => $role ), 'and', 'name' );
```

Comment:
Do we have a get types sort of function? I see we are always getting it from the bp global...

Reply:
We already have `bp_activity_get_types()` See https://github.com/buddypress/buddypress/blob/54d3c6140bf0a773f06032f8bf9eb112404d3fe7/src/bp-activity/bp-activity-functions.php#L886